### PR TITLE
Update headings to gem

### PR DESCRIPTION
**Pull request by chris-gds**
## What?

Changing headings to gem components

## Why?

To align architecture, improve consistency and in turn improve a11y.

## Visuals

Links

[`/government/fields-of-operation/iraq`](http://www.gov.uk/government/fields-of-operation/iraq)
[`/world/embassies`](http://www.gov.uk/world/embassies)
[`/government/organisations/medicines-and-healthcare-products-regulatory-agency/email-signup`](http://gov.uk/government/organisations/medicines-and-healthcare-products-regulatory-agency/email-signup)

(Before - left, After - right)

Screenshot 2021-04-13 at 09 53 14

Screenshot 2021-04-13 at 09 52 09

Screenshot 2021-04-07 at 09 35 08

Screenshot 2021-04-07 at 09 34 27

image

image

## Anything else?

The vast majority of headings appears in sections in Whitehall that are due for Redesign or are within the internal user facing _admin_ section of Whitehall both of which has been deprioritised for component swaps.

These include headings that appear under:

`/world/france/news`
`/government/topical-events/cop26`
`/government/how-government-works`
`/government/get-involved`
`/government/fields-of-operation`
`/government/organisations/`
`/government/organisations/charity-Commission/about`
`/government/history/past-prime-ministers/william-pitt`
`/government/history/past-foreign-secretaries/william-grenville`
`/government/latest?page=2&world_locations%5B%5D=singapore`

&
`/views/admin`

## Also 

[`/world/embassies`](http://www.gov.uk/world/embassies)
Has had a first pass refactor to apply the [Design System](https://design-system.service.gov.uk/) to the page, however many a11y issues remain. This is out of scope of this PR. 

Notable changes:
- The visual link style has changed slightly 
- The heading "A-Z" is not longer hidden on mobile.

### CODE REVIEW

**Comment by owenatgov** [comment]
Tiny nitpick before I start the review: in the PR description, the 2nd page reference links to the Iraq page. For the purpose of documentation could this be fixed please?

File: app/views/operational_fields/show.html.erb [hunk 1]
```diff
@@ -21,7 +21,6 @@
     </div>
   </div>
 </header>
-
```
Comment (owenatgov) [inline app/views/operational_fields/show.html.erb:24]: Non-blocker: Are all these space removals coming from a linter in your editor or something? It's unnecessary to remove them as rails compiles the html on build, these are more for readability.

File: app/views/operational_fields/show.html.erb [hunk 2]
```diff
@@ -66,15 +68,15 @@
                   </h3>
                 <% end %>
                 <ul class="casualties">
-                <% if fatality_notice.fatality_notice_casualties.any? %>
+                  <% if fatality_notice.fatality_notice_casualties.any? %>
```
Comment (owenatgov) [inline app/views/operational_fields/show.html.erb:71]: Yay tab structure ✨

**Review by owenatgov — COMMENTED** [review]
Couple of comments. Looking good :)

File: app/views/operational_fields/show.html.erb (hunk 1 above)
Comment (chris-gds) [reply app/views/operational_fields/show.html.erb]: Yeah exactly - part of a beautifier to automate formatting. I've restored, empty line settings need a tweak.

File: app/assets/stylesheets/frontend/views/_embassies.scss [hunk 3]
```diff
@@ -29,7 +29,7 @@
 
     @include media(tablet) {
       width: 82%;
-      margin: -50px 0 0 18%;
+      margin: -40px 0 0 18%;
```
[the remark this answers is not in the document]
Comment (chris-gds) [reply app/assets/stylesheets/frontend/views/_embassies.scss]: So, taking this feedback on board, I've refactored this page to apply a _first pass_ of the Design System.  It's changed the visuals slightly (documented on updated PR)

Taking a critical look at this page from an accessibility POV - it's _better_ but it needs work.  Screen reader testing reveals various problems. I'll be noting the issues but i'm mindful of scope of this ticket, so it's better but there's _lots_ of room improvement.

File: app/views/embassies/index.html.erb [hunk 4]
```diff
@@ -9,32 +8,35 @@
...
   <div class="inner-block embassy-listing">
-
-    <span class="a-z-heading"><span class="visuallyhidden">Countries ordered from </span><h1>A&ndash;Z</h1></span>
-
+    <span class="a-z-heading"><span class="visuallyhidden">Countries ordered from </span>
+      <%= render "govuk_publishing_components/components/heading", {
+        text: "A–Z",
+        font_size: "l"
+      } %>
+    </span>
```
[the remark this answers is not in the document]
Comment (chris-gds) [reply app/views/embassies/index.html.erb]: Nice. Good point, updated - thanks for the snippet also 🙌🏼

File: app/assets/stylesheets/frontend/views/_embassies.scss (hunk 3 above)
[the remark this answers is not in the document]
Comment (owenatgov) [reply app/assets/stylesheets/frontend/views/_embassies.scss]: Completely conscious of the scope of this ticket. Whitehall's grid system is a bit of a spooky one. This is looking really nice now!

File: app/views/embassies/index.html.erb [hunk 5]
```diff
@@ -9,33 +8,50 @@
...
-              <%= render partial: "organisation", collection: embassy.consular_services_organisations -%>
-            <% end %>
-          </ul>
-        <% end -%>
-      </li>
-      <% end -%>
-    </ol>
+<div class="govuk-main-wrapper">
+  <div class="govuk-grid-row">
+    <div class="embassy-listing">
```
Comment (owenatgov) [inline app/views/embassies/index.html.erb:13]: 2. I can't see really what this is doing, even after reading the stylesheet above. Removing this class doesn't change anything for me. I reckon you can remove this.

**Review by owenatgov — COMMENTED** [review]
2 final comments. I appreciate that this is getting beyond the scope of the card but I don't _think_ it's going to impact this work too much. Just a little tidy up.

**Review by owenatgov — APPROVED** [review]
Looks fab! Top work Chris

**Pull request merged**